File: Desktop/Aptiv/Management/src/main/java/com/Management/model/Department.java
package com.Management.model;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table(name="department")

public class Department { 

	@javax.persistence.Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="Id", nullable=false, unique=true, length=11)
	public int Id;
	
	@OneToMany(mappedBy = "department", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Theme> themes;
	
	//the department class knows only its manager
	@OneToOne(mappedBy = "department", fetch = FetchType.LAZY,
            cascade = CascadeType.ALL)
    private User user;
}

File: Desktop/Aptiv/Front/src/app/classes/department.ts
import { Theme } from "./theme";
import { User } from "./user";

export class Department {

    id?:bigint;
    name!:String;
    themes!:Theme[];
    users!:User[];
    coordinator:String="";
    manager:String="";
    gmanager:String="";
    coordinatorEmail:String="";
    managerEmail:String="";
    gmanagerEmail:String="";
}
